### ISSUE TITLE
option narg does not behave as documented

### ISSUE THREAD

**Issue by bschick** [yargs/yargs#2442]
### Problem

nargs options config does not consume values starting with `-` as documented

### Expected Results
The docs describe nargs to work like this:
```
var argv = require('yargs/yargs')(process.argv.slice(2))
  .nargs('token', 1)
  .parse(['--token', '-my-token']);
```
> parses as:
> { _: [], token: '-my-token', '$0': 'node test' }

### Actual Result
yargs 17.7.2 instead does not consume the `-my-token` and returns an error:

> Options:
>   --help     Show help                                                 [boolean]
>   --version  Show version number                                       [boolean]
> 
> Not enough arguments following: token

**Comment by shadowspawn** [comment]
You are right. The documentation is out of date. The default behaviour in `yargs-parser` changed quite a while ago and the documentation in `yargs` has not been updated.

- https://github.com/yargs/yargs-parser/pull/114
- https://github.com/yargs/yargs-parser/pull/246
- https://github.com/yargs/yargs-parser#nargs-eats-options
